pkg/agent/multicast/mcast_discovery.go

```diff
@@ -97,14 +99,48 @@ func (s *IGMPSnooper) queryIGMP(group net.IP, versions []uint8) error {
 		if err != nil {
 			return err
 		}
-		if err := s.ofClient.SendIGMPQueryPacketOut(igmpQueryDstMac, mcastAllHosts, openflow13.P_NORMAL, igmp); err != nil {
+		outPort := uint32(0)
+		if err := s.ofClient.SendIGMPQueryPacketOut(igmpQueryDstMac, types.McastAllHosts, outPort, igmp); err != nil {
 			return err
 		}
-		klog.V(2).InfoS("Sent packetOut for IGMP query", "group", group.String(), "version", version)
+		klog.V(2).InfoS("Sent packetOut for IGMP query", "group", group.String(), "version", version, "outPort", outPort)
 	}
 	return nil
 }
 
+func (s *IGMPSnooper) validate(event *mcastGroupEvent) (bool, error) {
+	if s.validator == nil {
```

Comment:
in case user disables acnp, while enabling multicast.

Reply:
I don't think it is a good idea to introduce both MulticastValidator and flag anpEnabled, and we can pass nil instead of networkPolicyController when ANP is diabled when calling `NewMulticastController` in cmd/antrea-agent/agent.go ?

Reply:
done